nrich-notification/README.md

```diff
@@ -102,39 +196,261 @@ public class NotificationTestController {
+public class ExampleController {
 
-content is resolved from code: `actionCode.content`
+    @GetMapping("manual")
+    public NotificationResponse manualExample() {
+        return notificationResponseService.responseWithNotification("manual.example");
+    }
 
-Action code is fully qualified class name when resolving notification for exception (i.e. `net.croz.nrich.notification.stub.NotificationResolverServiceTestException`) or a fixed string. For validation
-failure notifications content and title are resolved always from fixed strings:
+}
+```
+
+With `messages.properties` file set as this:
+
+```properties
+manual.example.title=Manual title
+manual.example.content=Manual content
+```
+
+response is:
+
+```json
+{
+    "notification": {
+        "title": "Manual title",
+        "content": "Manual content",
+        "messageList": [],
+        "severity": "INFO",
+        "uxNotificationOptions": null,
+        "timestamp": "2022-07-19T08:22:01.979151Z"
+    }
+}
+```
 
-`notification.validation-failed.title` and `notification.validation-failed.content`
+#### Automatic resolving
 
-Validation failure errors are added as messages to notification instances. When validation fails for `net.croz.nrich.notification.stub.NotificationResolverServiceTestRequest` instance on field  `name`
-with constraint `NotNull` validation failure messages are resolved from following codes:
+Automatic resolving of the notification action code was used in chapter describing the action notification.
+Method `responseWithNotificationActionResolvedFromRequest` will resolve the action code from current request
+and the key for resolving title and content is: `notification-example.save`.
 
-- `net.croz.nrich.notification.stub.NotificationResolverServiceTestRequest.name.NotNull.invalid`
-- `notificationResolverServiceTestRequest.name.NotNull.invalid`
-- `name.NotNull.invalid`
+If we were to put these key-value pairs into the `messages.properties` file:
+
+```properties
+notification-example.save.title=Automatic resolution title
+notification-example.save.content=Automatic resolution content
+```
+
+then the response that we would receive would be different from the one already seen:
+
+```json
+{
+    "notification": {
+        "title": "Automatic resolution title",
+        "content": "Automatic resolution content",
+        "messageList": [],
+        "severity": "INFO",
+        "uxNotificationOptions": null,
+        "timestamp": "2022-07-19T08:33:32.800894Z"
+    },
+    "data": {
+        "value": "Hello there!"
+    }
+}
+```
+
+### Custom validation notification data
+
+For validation failure notifications, content and title are always resolved from fixed strings:
+
+- `notification.validation-failed.title`
+- `notification.validation-failed.content`
+
+Validation failure errors are added as messages to notification instances and their failure messages can also be customized.
+
+For example, class situated in package `example`:
```

Comment:
Maybe just put the package statement in the class example?

Reply:
I put it into the class, but i would still leave it explicitly in text because package name can be overlooked in code.

Reply:
Ok